test_separate_tables failure with Pillow 11.2.1

Hi, I am in a process of updating pillow we deliver with Oracle Solaris to version 11.2.1, and I am hitting the following failure:

```pytb
===================================================== FAILURES ======================================================
_________________________________________ TestFileJpeg.test_separate_tables _________________________________________

self = <Tests.test_file_jpeg.TestFileJpeg object at 0x7fd12ce1a8a0>

    def test_separate_tables(self) -> None:
        im = hopper()
        data = []  # [interchange, tables-only, image-only]
        for streamtype in range(3):
            out = BytesIO()
            im.save(out, format="JPEG", streamtype=streamtype)
            data.append(out.getvalue())

        # SOI, EOI
        for marker in b"\xff\xd8", b"\xff\xd9":
            assert marker in data[1]
            assert marker in data[2]
        # DHT, DQT
        for marker in b"\xff\xc4", b"\xff\xdb":
>           assert marker in data[1]
E           AssertionError: assert b'\xff\xc4' in b'\xff\xd8\xff\xdb\x00C\x00\x08\x06\x06\x07\x06\x05\x08\x07\x07\x07\t\t\x08\n\x0c\x14\r\x0c\x0b\x0b\x0c\x19\x12\x13\x0....342\x
ff\xdb\x00C\x01\x08\t\t\x0c\x0b\x0c\x18\r\r\x182!\x1c!22222222222222222222222222222222222222222222222222\xff\xd9'
```

Only the `b"\xff\xc4"` marker is problematic - when I remove it, the test passes again.

I am getting the same error with all Pythons we have (3.9, 3.11 and 3.13) and with both libjpeg 9e and 9f.

Unfortunately, I am unsure what might be wrong here. Would you have some pointers/suggestions as for where to look?

### What are your OS, Python and Pillow versions?

* OS: Oracle Solaris
* Python: 3.9.22, 3.11.12, 3.13.3
* Pillow: 11.2.1

```text
--------------------------------------------------------------------
Pillow 11.2.1
Python 3.13.3 (main, May 16 2025, 09:09:42) [GCC 14.2.0]
--------------------------------------------------------------------
Python executable is /usr/bin/python3.13
System Python files loaded from /usr
--------------------------------------------------------------------
Python Pillow modules loaded from /userland-gate/components/python/pillow/build/prototype/i386/usr/lib/python3.13/vendor-packages/PIL
Binary Pillow modules loaded from /userland-gate/components/python/pillow/build/prototype/i386/usr/lib/python3.13/vendor-packages/PIL
--------------------------------------------------------------------
--- PIL CORE support ok, compiled for 11.2.1
*** TKINTER support not installed
--- FREETYPE2 support ok, loaded 2.13.3
--- LITTLECMS2 support ok, loaded 2.16
--- WEBP support ok, loaded 1.3.2
*** AVIF support not installed
--- JPEG support ok, compiled for 9.0
--- OPENJPEG (JPEG2000) support ok, loaded 2.5.3
--- ZLIB (PNG/ZIP) support ok, loaded 1.3.1
--- LIBTIFF support ok, loaded 4.6.0
*** RAQM (Bidirectional Text) support not installed
*** LIBIMAGEQUANT (Quantization method) support not installed
--- XCB (X protocol) support ok
--------------------------------------------------------------------
```

--- a/Tests/test_file_jpeg.py
+++ b/Tests/test_file_jpeg.py
@@ -1067,10 +1067,16 @@ class TestFileJpeg:
         for marker in b"\xff\xd8", b"\xff\xd9":
             assert marker in data[1]
             assert marker in data[2]
-        # DHT, DQT
-        for marker in b"\xff\xc4", b"\xff\xdb":
+
+        # DQT
+        markers = [b"\xff\xdb"]
+        if features.check_feature("libjpeg_turbo"):
+            # DHT
+            markers.append(b"\xff\xc4")
+        for marker in markers:
             assert marker in data[1]
             assert marker not in data[2]
+
         # SOF0, SOS, APP0 (JFIF header)
         for marker in b"\xff\xc0", b"\xff\xda", b"\xff\xe0":
             assert marker not in data[1]

